build/stf-run-ci/tasks/setup_stf_local_build.yml

```diff
@@ -89,6 +89,7 @@
   - name: Load Service Telemetry Operator RBAC
     ansible.builtin.command:
       cmd: oc apply -f ../deploy/{{ item }} -n "{{ namespace }}"
+      chdir: "{{ base_dir }}"
```

Comment:
This doesn't look right to me. Isn't the `../deploy/` directory relative to `{{ base_dir }}/working/smart-gateway-operator/build`, not relative to `base_dir`?

Reply:
@csibbitt This is the one for service-telemetry-operator.

base_dir is "{{ sto_dir }}/build"

With jenkins, the {{ base_dir }}/run-ci.yaml script is used to import the stf-run-ci role.
With zuul, the stf-run-ci role is imported from {{ sto_dir }}/ci/deploy_stf.yml playbook.

base_dir needs to be added here so that the path is not relative and doesn't break when importing the role from a different playbook.

Previously, this was only ever imported from run-ci.yaml, and some tasks used base_dir, while others did not. This inconsistency caused errors when trying to import the role in {{ sto_dir }}/ci/

base_dir is a weird value because it has been around a long time.
Ideally, IMHO everything should be relative to sto_dir, so that we can point at a particular instance of the repo and run everything from there. This change will be incorporated when I take a look at skipping cloning, which is needed for zuul, and should see a <project>_dir var added to define where the repos are cloned to/where to access the projects.

Reply:
Aha! It was indeed the STO vs. SGO that caught me. I understand now, thanks.

Reply:
Thanks for the detailed explanation Emma!